### ISSUE TITLE
Namespacing issue with libraries/helpers.rb

### ISSUE THREAD

**Issue by brint** [alertlogic/al_agents#25]
Howdy!

**TL;DR**: `libraries/helpers.rb` needs to be namespaced.

We were attempting to use this cookbook in conjunction with others, and we noticed that `helpers.rb` isn't namespaced which causes issues with other cookbooks. I've made a simple example cookbook in the following repo:

https://github.com/brint/libexample

Run `kitchen test` to see the trace. I've included a sample trace at the bottom of this issue.

The [service_name](https://github.com/alertlogic/al_agents/blob/master/libraries/helpers.rb#L14-L16) function was the one that bit us. We started noticing `al-agent` showing up as the `service_name` with the mysql cookbook (~> 5.0). In the mysql cookbook, `service_name` is a variable, and the helper function in this cookbook trumps the variable used in the library in their cookbook. Here's the specific line where the al_agents `service_name` function is causing issues within the mysql cookbook: 

https://github.com/chef-cookbooks/mysql/blob/v5.6.3/libraries/provider_mysql_service_rhel.rb#L123

Example trace using the libexample cookbook whose recipe only has a line of `include_recipe 'mysql::server'` and dependencies of mysql and al_agents:

```
       Recipe: mysql::server
       
           
           ================================================================================
           Error executing action `create` on resource 'mysql_service[default]'
           ================================================================================
           
           Chef::Exceptions::ResourceNotFound
           ----------------------------------
           resource template[/etc/my.cnf] is configured to notify resource service[al-agent] with action restart, but service[al-agent] cannot be found in the resource collection. template[/etc/my.cnf] is defined in /tmp/kitchen/cache/cookbooks/mysql/libraries/provider_mysql_service_rhel.rb:101:in `block in <class:Rhel>'
           
           Resource Declaration:
           ---------------------
           # In /tmp/kitchen/cache/cookbooks/mysql/recipes/server.rb
           
            20: mysql_service node['mysql']['service_name'] do
            21:   version node['mysql']['version']
            22:   port node['mysql']['port']
            23:   data_dir node['mysql']['data_dir']
            24:   server_root_password node['mysql']['server_root_password']
            25:   server_debian_password node['mysql']['server_debian_password']
            26:   server_repl_password node['mysql']['server_repl_password']
            27:   allow_remote_root node['mysql']['allow_remote_root']
            28:   remove_anonymous_users node['mysql']['remove_anonymous_users']
            29:   remove_test_database node['mysql']['remove_test_database']
            30:   root_network_acl node['mysql']['root_network_acl']
            31:   package_version node['mysql']['server_package_version']
            32:   package_action node['mysql']['server_package_action']
            33:   enable_utf8 node['mysql']['enable_utf8']
            34:   action :create
            35: end
           
           Compiled Resource:
           ------------------
           # Declared in /tmp/kitchen/cache/cookbooks/mysql/recipes/server.rb:20:in `from_file'
           
           mysql_service("default") do
             action [:create]
             retries 0
             retry_delay 2
             default_guard_interpreter :default
             declared_type :mysql_service
             cookbook_name "mysql"
             recipe_name "server"
             port "3306"
             data_dir "/var/lib/mysql"
             server_root_password "mysql_test"
             server_debian_password "gnuslashlinux4ev4r"
             remove_anonymous_users true
             remove_test_database true
             package_action "install"
           end
           
       
       Running handlers:
       [2015-08-12T02:06:23+00:00] ERROR: Running exception handlers
       Running handlers complete
       [2015-08-12T02:06:23+00:00] ERROR: Exception handlers complete
       Chef Client failed. 0 resources updated in 3.364878055 seconds
       [2015-08-12T02:06:23+00:00] FATAL: Stacktrace dumped to /tmp/kitchen/cache/chef-stacktrace.out
       [2015-08-12T02:06:23+00:00] ERROR: mysql_service[default] (mysql::server line 20) had an error: Chef::Exceptions::ResourceNotFound: resource template[/etc/my.cnf] is configured to notify resource service[al-agent] with action restart, but service[al-agent] cannot be found in the resource collection. template[/etc/my.cnf] is defined in /tmp/kitchen/cache/cookbooks/mysql/libraries/provider_mysql_service_rhel.rb:101:in `block in <class:Rhel>'
       
       [2015-08-12T02:06:24+00:00] FATAL: Chef::Exceptions::ChildConvergeError: Chef run process exited unsuccessfully (exit code 1)
```

**Comment by infernokodiak** [comment]
Thanks brint for the thorough writeup. The issue is acknowledged we'll be working on resolving as soon as possible.

**Comment by brint** [comment]
Thanks @infernokodiak!

**Comment by infernokodiak** [comment]
@brint, I updated the al_agents cookbooks to use namespaced helpers in a feature branch (https://github.com/alertlogic/al_agents/tree/feature/namespace-helpers).  I've updated your libexample to use my updated commit reference by changing the Berksfile.

https://github.com/infernokodiak/libexample

It appears to be working on the platforms I've tested and converged, save windows 2008 which at the moment I believe has nothing to do with the issue at hand but rather a winrm error.  If you'd like to verify please go ahead.  If it looks good I'll put in a PR.

Thanks!

**Comment by brint** [comment]
@infernokodiak Awesome, I'll give that branch a run.

**Comment by brint** [comment]
LGTM! :+1: When it's merged, could you guys push a release to supermarket as well? Thanks for the quick turn around!

**Comment by infernokodiak** [comment]
The issue has been fixed.  And a new version has been released (1.0.3)
